### PR TITLE
Fix logged event in response exception handler

### PR DESCRIPTION
**Pull request by doncem**
Report log events to AppInsights with correct implementation avoiding "wrong implementation" errors magically appearing.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/exception/UnauthenticatedException.java [hunk 1]
```diff
@@ -11,4 +11,13 @@
     public UnauthenticatedException(String message) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/sendletter/exception/UnauthenticatedException.java]: Do we need this constructor or can we remove it ?

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/exception/UnauthenticatedException.java]: This is used for missing header

File: src/main/java/uk/gov/hmcts/reform/sendletter/ResponseExceptionHandler.java [hunk 2]
```diff
@@ -59,7 +61,7 @@ protected void activateDirectFieldAccess(DataBinder dataBinder) {
 
     @ExceptionHandler(InvalidTokenException.class)
     protected ResponseEntity<Void> handleInvalidTokenException(InvalidTokenException exc) {
-        log.warn(exc.getMessage(), exc);
+        log.warn(exc.getMessage(), new UnauthenticatedException(exc));
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/ResponseExceptionHandler.java]: can we make `InvalidTokenException` extend the required base class instead?

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/ResponseExceptionHandler.java]: Comes from library

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/ResponseExceptionHandler.java]: let's change it :)

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/ResponseExceptionHandler.java]: 🎉

**Pull request merged**